File: ruby/FizzBuzzTest.rb
require 'test/unit'
require './FizzBuzz'

class FizzBuzzTest < Test::Unit::TestCase
    def setup 
        @obj = FizzBuzz.new
    end

    def test_1を入れたら1が返る
        assert_equal 1, @obj.run(1)
    end
    def test_2を入れたら2が返る
        assert_equal 2, @obj.run(2)
    end
    def test_3を入れたらFizzが返る
        assert_equal 'Fizz', @obj.run(3)
    end
    def test_6を入れたらFizzが返る
        assert_equal 'Fizz', @obj.run(6)
    end
    def test_5を入れたらBuzzが返る

    end
    def test_10を入れたらBuzzが返る

    end
    def test_15を入れたらFizzBuzzが返る

    end
    def test_30を入れたらFizzBuzzが返る

    end
end

File: haskell/FizzBuzzTest.hs
import Test.HUnit
import FizzBuzz 

tests = [
        "1を入れたら'1'を返す" ~: fizzbuzz 1 ~?= "1",
        "2を入れたら'2'を返す" ~: fizzbuzz 2 ~?= "2",
        "3を入れたら'Fizz'を返す" ~: fizzbuzz 3 ~?= "Fizz",
        "6を入れたら'Fizz'を返す" ~: fizzbuzz 6 ~?= "Fizz",
        "5を入れたら'Buzz'を返す" ~: fizzbuzz 5 ~?= "Buzz",
        "10を入れたら'Buzz'を返す" ~: fizzbuzz 10 ~?= "Buzz",
        "15を入れたら'FizzBuzz'を返す" ~: fizzbuzz 15 ~?= "FizzBuzz",
        "30を入れたら'FizzBuzz'を返す" ~: fizzbuzz 30 ~?= "FizzBuzz"
    ]

main::IO ()
main = do
    counts <- runTestTT $ TestList tests
    return ()